src/test/java/io/managed/services/test/cluster/DataPlaneClusterTest.java

```diff
@@ -237,49 +239,35 @@ public void testStandardKafkaNodeAutoscaling() {
 
         //take a snapshot of machine sets with name (INITIAL)
         Map<String, Integer> machineSetToReadyNodeCountSnapshotBefore =  FleetshardUtils.getReadyNodesPerEachMachineSetContainingName(oc, "kafka-standard");
-        log.info(machineSetToReadyNodeCountSnapshotBefore);
-
-        // obtain info how many remaining standard streaming units there are according to capacity
-        int remainingStandardInstancesCapacity = FleetshardUtils.getCapacityRemainingUnitsFromMKAgent(oc, ManagedKafkaType.standard);
-        log.info("currently remaining standard managed kafka instances capacity '{}'", remainingStandardInstancesCapacity);
-
-        // there is not enough space in the cluster to observe nodes scaling.
-        if (remainingStandardInstancesCapacity == 0)
-            throw new SkipException("Already reached max limit of standard instances created");
-
-        // creating one instance less than to reach max as that would not help observing max node creation anyway.
-        log.info("creating at least '{}' standard managed kafka instances", remainingStandardInstancesCapacity);
-
-        // wait up to 15 minutes for observing increase in nodes number in relevant machine sets, or creation of new instance
-        AtomicInteger placeholderInstanceSuffixCounter = new AtomicInteger(1);
+        log.debug(machineSetToReadyNodeCountSnapshotBefore);
+        // find out the minimum nodes value per machine set
+        int minNodeInMachineSet =  machineSetToReadyNodeCountSnapshotBefore.values().stream().mapToInt(Integer::valueOf).min().orElse(0);
+        log.info("currently minimum existing nodes in machine set '{}'", minNodeInMachineSet);
+
+        // get remaining capacity from MK agent
+        int remainingCapacity = FleetshardUtils.getCapacityRemainingUnitsFromMKAgent(oc, ManagedKafkaType.standard);
+        log.info("remaining capacity in the cluster '{}'", remainingCapacity);
+
+        // if either capacity or number of existing MK resources (type standard) already indicate full capacity there is no chance of obtaining correct  data within 15 minutes thus skip
+        if (remainingCapacity == 0 || minNodeInMachineSet == maxStandardStreamingUnitsInCluster)
+            throw new SkipException("cluster already reported to be at its maximum capacity");
```

Comment:
```suggestion
        if (remainingCapacity == 0)
            throw new SkipException("cluster already reported to be at its maximum capacity");
```

Reply:
`minNodeInMachineSet == maxStandardStreamingUnitsInCluster` could be true because of 6 real kafka + 1 reserved streaming unit.

I think checking for remainingCapacity is enough.